### PR TITLE
Update to latest versions and make it work on embedded hardware

### PR DESCRIPTION
**Pull request by jkristell**
* Updated to edition 2018
* Updated embedded-hal pin v2 api
* removed reset, and added enable and disable methods
* Added destroy method
* Added a delay as an argument to methods that generates clock

This code is tested on a Bluepill board with a 5kg load cell

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -5,17 +5,19 @@
 //! [`embedded-hal`]: https://docs.rs/embedded-hal/0.2
 
 #![deny(missing_docs)]
-#![deny(warnings)]
```
Comment (jonas-hagen) [inline src/lib.rs]: Why is this needed?

File: src/lib.rs [hunk 2]
```diff
@@ -30,62 +32,87 @@ pub struct Hx711<IN, OUT> {
...
+        DELAY: DelayUs<u16>,
+    {
+        self.pd_sck.set_low()?;
+        if self.dout.is_high()? {
             // Conversion not ready yet
             return Err(nb::Error::WouldBlock);
         }
 
+        // Dout falling -> clock high > 0.1 us 
+        delay.delay_us(1);
```
Comment (jonas-hagen) [inline src/lib.rs]: Using delay makes the program yield in linux-embedded-hal (RaspberryPi). This makes the delay longer than the maximum specified 50 us. Maybe this has this been fixed in recent updates to linux-embedded-hal?

Just cycling the pin is not necessarily better, but maybe you have an Idea on how to make it work on both, Linux and HW?

File: src/lib.rs [hunk 3]
```diff
@@ -30,62 +32,87 @@ pub struct Hx711<IN, OUT> {
...
-        }
-        self.pd_sck.set_low();
+    /// Enable 
+    pub fn enable(&mut self) -> Result<(), PINERR> {
+        self.pd_sck.set_low()
+    }
+
+    /// Disable - after 60 us of clk high the chip will go to sleep
+    pub fn disable(&mut self) -> Result<(), PINERR> {
+        self.pd_sck.set_high()
```
Comment (jonas-hagen) [inline src/lib.rs]: From the datasheet:
> After a reset or power-down event, input selection is default to Channel A with a gain of 128.

So we need a delay here to have a reproducible state for the next enable.

File: src/lib.rs [hunk 4]
```diff
@@ -30,62 +32,87 @@ pub struct Hx711<IN, OUT> {
...
-    /// Reset the chip. Mode is Channel A Gain 128 after reset.
-    pub fn reset(&mut self) {
-        self.pd_sck.set_high();
-        for _ in 1..3 {
-            self.dout.is_high();
-        }
-        self.pd_sck.set_low();
+    /// Enable 
+    pub fn enable(&mut self) -> Result<(), PINERR> {
+        self.pd_sck.set_low()
```
Comment (jonas-hagen) [inline src/lib.rs]: Set mode after enable, because sleep mode resets the chip.

**Review by jonas-hagen — COMMENTED** [review]
Thanks for your work! (I was off-grid the last two weeks)

The following changes look good to me (and if they are in a separate commit, I will merge them):
- Updated to edition 2018
- Updated embedded-hal pin v2 api
- Added destroy method

The following changes bring up the discussion about delays, and I would like to test that with the linux-embedded-hal before merging (see comments in the code):
- removed reset, and added enable and disable methods
- Added a delay as an argument to methods that generates clock

The following change is missing:
- Add "Tested on STM32F103 (BluePill)" to the README, preferably with a link to a repo with any (!) example code. 

Regarding delays: Getting a delay between 0.2 and 50 us (as required) is difficult on linux and using the embedded-hal delay resulted in delays being always longer than 50 us. The best way I found was just to cycle the pins and use `count <<= 1;` as delay. Maybe you have good ideas how to approach this in documentation or code? Dropping support for (standard) linux is also an option!

**Comment by jkristell** [comment]
Hi and thanks for the review!

About the delays and embedded-linux-hal:

One solution to keeping the driver working with both embedded-hal and embedded-hal-linux could be to wrap the delayified version of the driver in a embedded-linux version with an internal delay that is just a no-op for delay_us(1) and some configurable delay time for the other delay needed. Still a hack but at least the driver could work for both scenarios without splitting the core logic into to drivers.

File: src/lib.rs (hunk 1 above)
Comment (jkristell) [reply src/lib.rs]: It's not needed as far as I know. Probably just got removed during the development. I will add it back!

File: src/lib.rs (hunk 3 above)
Comment (jkristell) [reply src/lib.rs]: Good point!

File: src/lib.rs (hunk 4 above)
Comment (jkristell) [reply src/lib.rs]: True! Didn't think about that.

**Comment by jkristell** [comment]
Regarding the example. The repo with that code is not public yet. It needs some cleanups, but I could do that.
